semantic_release/helpers.py

```diff
@@ -73,67 +74,94 @@ def dynamic_import(import_path: str) -> Any:
+        git@github.com:username/myproject.git                  <-- assumes ssh://
+        ssh://git@github.com:3759/myproject.git                <-- non-standard, but assume user 3759
+        ssh://git@github.com:username/myproject.git
+        ssh://git@bitbucket.org:7999/username/myproject.git
+        git+ssh://git@github.com:username/myproject.git
+        /Users/username/dev/remote/myproject.git               <-- Posix File paths
+        file:///Users/username/dev/remote/myproject.git
+        C:/Users/username/dev/remote/myproject.git             <-- Windows File paths
+        file:///C:/Users/username/dev/remote/myproject.git
+
+    REFERENCE: https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats
+
     Raises ValueError if the url can't be parsed.
     """
     log.debug("Parsing git url %r", url)
+
+    # Normalizers are a list of tuples of (pattern, replacement)
+    normalizers = [
+        # normalize implicit ssh urls to explicit ssh://
+        (r"^(\w+@)", r"ssh://\1"),
+
+        # normalize git+ssh:// urls to ssh://
+        (r"^git\+ssh://", "ssh://"),
+
+        # normalize an scp like syntax to URL compatible syntax
+        # excluding port definitions (:#####) & including numeric usernames
+        (r"(ssh://(?:\w+@)?[\w.]+):(?!\d{1,5}/\w+/)(.*)$", r"\1/\2"),
+
+        # normalize implicit file (windows || posix) urls to explicit file:// urls
+        (r"^([C-Z]:/)|^/(\w)", r"file:///\1\2"),
+    ]
+
+    for pattern, replacement in normalizers:
+        url = re.compile(pattern).sub(replacement, url)
+
+    # run the url through urlsplit to separate out the parts
     urllib_split = urlsplit(url)
-    if urllib_split.scheme:
-        # We have been able to parse the url with urlsplit,
-        # so it's a (git|ssh|https?)://... structure
-        namespace, _, name = urllib_split.path.lstrip("/").rpartition("/")
-        name.rstrip("/")
-        name = name[:-4] if name.endswith(".git") else name
-        if not all((urllib_split.scheme, urllib_split.netloc, namespace, name)):
-            raise ValueError(f"Bad url: {url!r}")
-
-        return ParsedGitUrl(
-            scheme=urllib_split.scheme,
-            netloc=urllib_split.netloc,
-            namespace=namespace,
-            repo_name=name,
-        )
 
-    m = GIT_URL_REGEX.match(url)
-    if not m:
+    # Fail if url scheme not found
+    if not urllib_split.scheme:
         raise ValueError(f"Cannot parse {url!r}")
 
-    repo_name = m.group("repo_name")
-    repo_name = repo_name[:-4] if repo_name.endswith(".git") else repo_name
+    # We have been able to parse the url with urlsplit,
+    # so it's a (file|git|ssh|https?)://... structure
+    # but we aren't validating the protocol scheme as its not our business
+
+    # use PosixPath to normalize the path & then separate out the namespace & repo_name
+    namespace, _, name = str(PosixPath(urllib_split.path)).lstrip("/").rpartition("/")
```

Comment:
Given we're just using this for string normalisation I think it's a bit clearer to use [`PurePosixPath`](https://docs.python.org/3.7/library/pathlib.html#pathlib.PurePosixPath), as `Pure*Path` classes don't make any syscalls

Reply:
I did not know that! I learned today! Thx!

Reply:
I only found that difference out myself pretty recently 😁